File: Examen/QytetetRuby/lib/tramposo.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
    attr_accessor :veces_trampa, :importe_fraudulento
    class Tramposo < Jugador
        def initialize(jugador)
            @encarcelado = jugador.encarcelado
            @nombre = jugador.nombre
            @saldo = jugador.saldo
            @carta_libertad = jugador.carta_libertad
            @casilla_actual = jugador.casilla_actual
            @propiedades = jugador.propiedades
            @veces_trampa = 0
            @importe_fraudulento = 0
        end
        
        def modificar_saldo(cantidad)
            valor = rand(2)
            
            if valor == 0
                super(cantidad/2)
                @veces_trampa = @veces_trampa + 1
                @importe_fraudulento = @importe_fraudulento + cantidad/2
           
            else if valor == 1
                 super(cantidad)   
            end
            end
        end
        
        def perdonar
            @veces_trampa = 0
            @importe_fraudulento = 0
        end
        
        def to_s
            super + "Tramposo { veces_trampa: #{@veces_trampa}, importe_fraudulento: #{@importe_fraudulento}"
        end
    end
end

File: Examen/modeloqytetet/src/ModeloQytetet/Tramposo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

//EXAMEN
package ModeloQytetet;

import java.util.Random;

/**
 *
 * @author sergioff98
 */
public class Tramposo extends Jugador {
    int vecesTrampa;
    double importeFraudulento;
    
    protected Tramposo(Jugador jugador)
    {
        super(jugador);
        vecesTrampa = 0;
        importeFraudulento = 0;
    }
    
    @Override
    public void modificarSaldo(int cantidad){
        Random generator=new Random();
        int numero = generator.nextInt(2);
        
        if(numero==0){
            super.modificarSaldo(cantidad/2);
            vecesTrampa = vecesTrampa + 1;
            importeFraudulento = importeFraudulento + cantidad/2;
        }
        else if(numero==1)
            super.modificarSaldo(cantidad);
    }
    
    public void perdonar()
    {
        vecesTrampa=0;
        importeFraudulento=0;
    }

    @Override
    public String toString() {
        return (super.toString() + "Tramposo{" + "vecesTrampa=" + vecesTrampa + ", importeFraudulento=" + importeFraudulento + '}');
    }
    
    
     
}
